kong/plugins/ddtrace/span.lua

```diff
@@ -74,18 +76,26 @@ function span_methods:set_sampling_priority(sampling_priority)
     self.metrics["_sampling_priority_v1"] = sampling_priority
 end
 
+function span_methods:set_tags(tags)
+    if type(tags) == "table" then
+        for k,v in pairs(tags) do
+            self:set_tag(k, v)
+        end
+    end
+end
 
 function span_methods:new_child(name, resource, start)
     return new(
-    self.service,
-    name,
-    resource,
-    self.trace_id,
-    generate_span_id(),
-    self.span_id,
-    start,
-    self.sampling_priority,
-    self.origin
+        self.service,
+        name,
+        resource,
+        self.trace_id,
+        generate_span_id(),
+        self.span_id,
+        start,
+        self.sampling_priority,
+        self.origin,
+        self
```

Comment:
If there are more than two spans, than `self` is not always the local root span.